packages/signals/signals/src/core/client/index.ts

```diff
@@ -8,13 +8,17 @@ export class SignalsIngestSettings {
   flushInterval: number
   apiHost: string
   shouldDisableSignalRedaction: () => boolean
+  signalIngestion: () => boolean
   writeKey?: string
+  sampleRate: number
   constructor(settings: SignalsIngestSettingsConfig) {
     this.flushAt = settings.flushAt ?? 5
     this.apiHost = settings.apiHost ?? 'signals.segment.io/v1'
     this.flushInterval = settings.flushInterval ?? 2000
     this.shouldDisableSignalRedaction =
       settings.shouldDisableSignalRedaction ?? (() => false)
+    this.signalIngestion = settings.signalIngestion ?? (() => false)
+    this.sampleRate = settings.sampleRate ?? 0
```

Comment:
I'd rather we not expose sampleRate publicly, as it's strictly an implementation detail -- IMO, you shouldn't be able to customize the sample rate as a setting. Can we abstract this way completely? in my head the `signalsIngestion` was the only data point we want